assets/src/edit-story/components/library/panes/media/mediaPane.js

```diff
@@ -182,69 +222,97 @@ function MediaPane(props) {
-            <Column>
-              {resources
-                .filter((_, index) => isEven(index))
-                .map((resource) => (
-                  <MediaElement
-                    resource={resource}
-                    key={resource.src}
-                    width={DEFAULT_WIDTH}
-                    onInsert={insertMediaElement}
-                  />
-                ))}
-            </Column>
-            <Column>
-              {resources
-                .filter((_, index) => !isEven(index))
-                .map((resource) => (
-                  <MediaElement
-                    resource={resource}
-                    key={resource.src}
-                    width={DEFAULT_WIDTH}
-                    onInsert={insertMediaElement}
-                  />
-                ))}
-            </Column>
-          </Container>
-        )}
+        <Inner>
+          <Padding>
+            <Header>
+              <Title>
+                {__('Media', 'web-stories')}
+                {(!isMediaLoaded || isMediaLoading) && <Spinner />}
+              </Title>
+              <MainButton onClick={openMediaPicker}>
+                {__('Upload', 'web-stories')}
+              </MainButton>
+            </Header>
+
+            <SearchInput
+              value={searchTerm}
+              placeholder={__('Search media...', 'web-stories')}
+              onChange={onSearch}
+            />
+
+            <FilterButtons>
+              {FILTERS.map(({ filter, name }, index) => (
+                <FilterButton
+                  key={index}
+                  active={filter === mediaType}
+                  onClick={onFilter(filter)}
+                >
+                  {name}
+                </FilterButton>
+              ))}
+            </FilterButtons>
+          </Padding>
+
+          {isMediaLoaded && !media.length ? (
+            <Message>{__('No media found', 'web-stories')}</Message>
+          ) : (
+            <Container ref={refContainer}>
+              <Column>
+                {resources
+                  .filter((_, index) => isEven(index))
+                  .map((resource, i) => (
+                    <MediaElement
+                      resource={resource}
+                      key={i}
+                      width={DEFAULT_WIDTH}
+                      onInsert={insertMediaElement}
+                    />
+                  ))}
+              </Column>
+              <Column>
+                {resources
+                  .filter((_, index) => !isEven(index))
+                  .map((resource, i) => (
+                    <MediaElement
+                      resource={resource}
+                      key={i}
+                      width={DEFAULT_WIDTH}
+                      onInsert={insertMediaElement}
+                    />
+                  ))}
+              </Column>
+              {hasMore && (
+                <div ref={refContainerFooter}>
+                  {__('Loading', 'web-stories')}
```

Comment:
Was there any specific UX for this? Probably for a follow-up PR?

Reply:
There is a follow up ticket -  https://github.com/google/web-stories-wp/issues/439